src/services/classifier2020.ts

```diff
@@ -225,6 +225,10 @@ namespace ts.classifier.v2020 {
         return (isQualifiedName(node.parent) && node.parent.right === node) || (isPropertyAccessExpression(node.parent) && node.parent.name === node);
     }
 
+    function isInfinityOrNaNString(name: __String): boolean {
+        return name === "Infinity" || name === "-Infinity" || name === "NaN";

```

Comment:
I think you need a more thorough check, but some quick feedback: there's no identifier named `-Infinity`.

Reply:
> I think you need a more thorough check

Do you have something in particular in mind? Not sure what else could be done as `Infinity` and `NaN` are parsed as `SyntaxKind.Identifier` and I also found this check used in `checker.ts`

Reply:
I've switched this PR to use the typechecker's function - which solves this 👍🏻